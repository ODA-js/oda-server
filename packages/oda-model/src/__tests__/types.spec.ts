import 'jest';

import {
  ArrayToHash,
  HashToArray,
  ArrayToMap,
  HashToMap,
  MapToArray,
  MapToHash,
} from '../types';

describe('helpers', () => {
  it('ArrayToMap', () => {
    expect(
      ArrayToMap([{ name: '1', other: '2' }, { name: '2', other: '2' }]),
    ).toMatchSnapshot('ArrayToMap');
  });
  it('HashToMap', () => {
    expect(
      HashToMap({
        1: {
          other: 1,
        },
        2: {
          other: 2,
        },
      }),
    ).toMatchSnapshot('HashToMap');
  });
  it('HashToArray', () => {
    expect(
      HashToArray({
        1: {
          other: 1,
        },
        2: {
          other: 2,
        },
      }),
    ).toMatchSnapshot('HashToArray');
  });
  it('ArrayToHash', () => {
    expect(
      ArrayToHash([{ name: '1', other: '2' }, { name: '2', other: '2' }]),
    ).toMatchSnapshot('ArrayToHash');
  });
  it('MapToHash', () => {
    expect(
      MapToHash(
        new Map([
          ['1', { name: '2', other: '1' }],
          ['3', { name: '3', other: 3 }],
        ]),
        value => ({ ...value, name: undefined }),
      ),
    ).toMatchSnapshot('MapToHash');
    expect(
      MapToHash(
        new Map([
          ['key 1', { name: '2', other: '1' }],
          ['key 2', { name: '3', other: 3 }],
        ]),
      ),
    ).toMatchSnapshot('MapToHash-Values');
  });
  it('MapToArray', () => {
    expect(
      MapToArray(
        new Map([
          ['1', { name: '2', other: '1' }],
          ['3', { name: '3', other: 3 }],
        ]),
        (key, value) => ({ ...value, name: key }),
      ),
    ).toMatchSnapshot('MapToArray');
    expect(
      MapToArray(
        new Map([
          ['key 1', { name: '2', other: '1' }],
          ['key 2', { name: '3', other: 3 }],
        ]),
      ),
    ).toMatchSnapshot('MapToArray-Values');
  });
});
