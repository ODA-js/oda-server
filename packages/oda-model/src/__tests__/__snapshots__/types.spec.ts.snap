// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers ArrayToHash: ArrayToHash 1`] = `
Object {
  "1": Object {
    "name": "1",
    "other": "2",
  },
  "2": Object {
    "name": "2",
    "other": "2",
  },
}
`;

exports[`helpers ArrayToMap: ArrayToMap 1`] = `
Map {
  "1" => Object {
    "name": "1",
    "other": "2",
  },
  "2" => Object {
    "name": "2",
    "other": "2",
  },
}
`;

exports[`helpers ArrayToMap: ArrayToMap with process 1`] = `
Map {
  "1" => Object {
    "key": "1",
    "name": "1",
    "other": "2",
  },
  "2" => Object {
    "key": "2",
    "name": "2",
    "other": "2",
  },
}
`;

exports[`helpers HashToArray: HashToArray 1`] = `
Array [
  Object {
    "name": "1",
    "other": 1,
  },
  Object {
    "name": "2",
    "other": 2,
  },
]
`;

exports[`helpers HashToArray: HashToArray with process 1`] = `
Array [
  Object {
    "name": "1",
    "other": 1,
  },
  Object {
    "name": "2",
    "other": 2,
  },
]
`;

exports[`helpers HashToMap: HashToMap 1`] = `
Map {
  "1" => Object {
    "name": "1",
    "other": 1,
  },
  "2" => Object {
    "name": "2",
    "other": 2,
  },
}
`;

exports[`helpers HashToMap: HashToMap with process 1`] = `
Map {
  "1" => Object {
    "name": "1",
    "other": 1,
  },
  "2" => Object {
    "name": "2",
    "other": 2,
  },
}
`;

exports[`helpers MapToArray: MapToArray 1`] = `
Array [
  Object {
    "name": "1",
    "other": "1",
  },
  Object {
    "name": "3",
    "other": 3,
  },
]
`;

exports[`helpers MapToArray: MapToArray-Values 1`] = `
Array [
  Object {
    "name": "2",
    "other": "1",
  },
  Object {
    "name": "3",
    "other": 3,
  },
]
`;

exports[`helpers MapToHash: MapToHash 1`] = `
Object {
  "1": Object {
    "name": undefined,
    "other": "1",
  },
  "3": Object {
    "name": undefined,
    "other": 3,
  },
}
`;

exports[`helpers MapToHash: MapToHash-Values 1`] = `
Object {
  "key 1": Object {
    "name": "2",
    "other": "1",
  },
  "key 2": Object {
    "name": "3",
    "other": 3,
  },
}
`;

exports[`helpers MapToHash: MapToHash-Values with process 1`] = `
Object {
  "key 1": Object {
    "name": "2",
    "other": "1",
  },
  "key 2": Object {
    "name": "3",
    "other": 3,
  },
}
`;
